Tests/Get-BMDeployment.Tests.ps1

```diff
@@ -212,9 +241,150 @@ Describe 'Get-BMDeployment' {
+        $deployment2 = GivenDeployment $build -Stage 'Testing'
+        $deployment3 = GivenDeployment $build2
+        WhenGettingBMDeployment -Release $script:releaseRelease.name
+        ThenShouldNotThrowErrors
+        ThenTotalDeploymentsReturned 1
+        ThenDeploymentShouldNotBeReturned $deployment
+        ThenDeploymentShouldNotBeReturned $deployment2
+        ThenDeploymentShouldBeReturned $deployment3
+    }
+
+    It 'should get deployment by build object' {
+        $build = GivenReleaseBuild $script:releaseAll
+        $build2 = GivenReleaseBuild $script:releaseRelease
+        $deployment = GivenDeployment $build
+        $deployment2 = GivenDeployment $build -Stage 'Testing'
+        $deployment3 = GivenDeployment $build2
+        WhenGettingBMDeployment -Build $build
+        ThenShouldNotThrowErrors
+        ThenTotalDeploymentsReturned 2
+        ThenDeploymentShouldBeReturned $deployment
+        ThenDeploymentShouldBeReturned $deployment2
+        ThenDeploymentShouldNotBeReturned $deployment3
+    }
+
+    It 'should get deployment by build id' {
+        $build = GivenReleaseBuild $script:releaseAll
+        $build2 = GivenReleaseBuild $script:releaseRelease
+        $deployment = GivenDeployment $build
+        $deployment2 = GivenDeployment $build -Stage 'Testing'
+        $deployment3 = GivenDeployment $build2
+        WhenGettingBMDeployment -Build $build.id
+        ThenShouldNotThrowErrors
+        ThenTotalDeploymentsReturned 2
+        ThenDeploymentShouldBeReturned $deployment
+        ThenDeploymentShouldBeReturned $deployment2
+        ThenDeploymentShouldNotBeReturned $deployment3
+    }
+
+    It 'should get deployment by pipeline name' {
+        $build = GivenReleaseBuild $script:releaseAll
+        $build2 = GivenReleaseBuild $script:releaseRelease
+        $deployment = GivenDeployment $build
+        $deployment2 = GivenDeployment $build -Stage 'Testing'
+        $deployment3 = GivenDeployment $build2
+        WhenGettingBMDeployment -PipelineName $script:pipeline.Pipeline_Name
+        ThenShouldNotThrowErrors
+        ThenTotalDeploymentsReturned 2
+        ThenDeploymentShouldBeReturned $deployment
+        ThenDeploymentShouldBeReturned $deployment2
+        ThenDeploymentShouldNotBeReturned $deployment3
+    }
+
+    It 'should get a deployment by pipeline stage name' {
+        $build = GivenReleaseBuild $script:releaseAll
+        $build2 = GivenReleaseBuild $script:releaseRelease
+        $deployment = GivenDeployment $build
+        $deployment2 = GivenDeployment $build -Stage 'Testing'
+        $deployment3 = GivenDeployment $build2
+        WhenGettingBMDeployment -PipelineStageName 'Testing'
+        ThenShouldNotThrowErrors
+        ThenDeploymentShouldNotBeReturned $deployment
+        ThenDeploymentShouldBeReturned $deployment2
+        ThenDeploymentShouldNetBeReturned $deployment3
+        ThenTotalDeploymentsReturned 1
+    }
+
+    It 'should get a deployment by status' {

```

Comment:
Add a test that:
* Tests getting deployments that match multiple parameters.
* Tests when no deployments are found when using multiple parameters.